orm_lib/src/SqlBinder.cc

```diff
@@ -266,22 +259,59 @@ SqlBinder &SqlBinder::operator<<(std::nullptr_t nullp)
     return *this;
 }
 
+SqlBinder &SqlBinder::operator<<(DefaultValue dv)
+{
+    (void)dv;
+    if (type_ == ClientType::PostgreSQL)
+    {
+        utils::replaceAll(*sqlPtr_,
+                          "$" + std::to_string(parametersNumber_ + 1),
+                          "default");
+
+        // decrement all other $n parameters by 1
+        int i = parametersNumber_ + 2;
+        while ((sqlPtr_->find("$" + std::to_string(parametersNumber_ + i))) !=
+               std::string::npos)
+        {
+            utils::replaceAll(*sqlPtr_,
+                              "$" + std::to_string(parametersNumber_ + i),
+                              "$" + std::to_string(parametersNumber_ + i - 1));
+            ++i;
+        }
+        sqlViewPtr_ = sqlPtr_->data();
+        sqlViewLength_ = sqlPtr_->length();
+    }
+    else if (type_ == ClientType::Mysql)
+    {
+        ++parametersNumber_;
+        parameters_.push_back(NULL);
+        lengths_.push_back(0);
+        formats_.push_back(DrogonDefaultValue);
+    }
+    else if (type_ == ClientType::Sqlite3)
+    {
+        LOG_FATAL << "default not supported in sqlite3";
+        exit(1);
```

Comment:
@an-tao Do we have a graceful way to handle such error? Maybe throw an exception?

Reply:
> @interfector18 Thanks for the PR! Pardon, I'm not familiar with this feature. May you describe a real-world use case?

When inserting a new recorder with some default values depend on runtime conditions, this interface is needed.

> 
> @an-tao How much works is needed to transition to using `PQsendPrepare`? I don't think there's a way around the replacement edge case (besides writing our own SQL parser). And using prepared statements should make drogon's queries faster.
> 

I didn't find a way to use PQSendPrepared with a default value instead of changing the SQL.

> I think some discussion will be needed.

Yes. I agree with u.

Reply:
> @an-tao Do we have a graceful way to handle such error? Maybe throw an exception?

I think this is an error of programing, throwing exceptions or exiting is okay.

Reply:
> I didn't find a way to use PQSendPrepared with a default value instead of changing the SQL.

Right, sorry. I misread the PR.
Well, I don't think there's a way around; unless someone wants to write a PgSQL parser?

Reply:
There is also the potential to improve performance, as you're modiying the sql string in one place instead of two (for column name vs just the $n or ? part). It also simplifies usage.

Reply:
Should be the same performance since the strings are small enough to fit in L1 cache